Client/Frontend/Home/FxPocketHomeHorizontalCell.swift

```diff
@@ -0,0 +1,196 @@
+                                                                       maxSize: FxHomeHorizontalCellUX.sponsoredFontSize)
+        label.textColor = .secondaryLabel
+        label.numberOfLines = 1
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = .FirefoxHomepage.Pocket.Sponsored
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private lazy var sponsoredIcon: UIImageView = {
+        let image = UIImageView(image: UIImage(named: ImageIdentifiers.sponsoredStar))
+        NSLayoutConstraint.activate([
+            image.heightAnchor.constraint(equalToConstant: 12),
+            image.widthAnchor.constraint(equalToConstant: 12)
+        ])
+        image.translatesAutoresizingMaskIntoConstraints = false
+        return image
+    }()
+
+    private lazy var descriptionLabel: UILabel = {
+        let label = UILabel()
+        label.adjustsFontForContentSizeCategory = true
+        label.font = DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .caption1,
+                                                                   maxSize: FxHomeHorizontalCellUX.siteFontSize)
+        label.textColor = .label
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    // MARK: - Variables
+    var notificationCenter: NotificationCenter = NotificationCenter.default
+
+    // MARK: - Inits
+
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+
+        applyTheme()
+        setupNotifications(forObserver: self,
+                           observing: [.DisplayThemeChanged])
+        setupLayout()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    deinit {
+        notificationCenter.removeObserver(self)
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        heroImageView.image = nil
+        descriptionLabel.text = nil
+        titleLabel.text = nil
+        applyTheme()
+    }
+
+    // MARK: - Helpers
+
+    func configure(viewModel: FxPocketHomeHorizontalCellViewModel) {
+        titleLabel.text = viewModel.title
+        descriptionLabel.text = viewModel.description
+
+        heroImageView.sd_setImage(with: viewModel.imageURL)
+        sponsoredStack.isHidden = viewModel.sponsor == nil
+        descriptionLabel.font = viewModel.sponsor == nil
+        ? DynamicFontHelper.defaultHelper.preferredFont(withTextStyle: .caption1,
+                                                        maxSize: FxHomeHorizontalCellUX.siteFontSize)
+        : DynamicFontHelper.defaultHelper.preferredBoldFont(withTextStyle: .caption1,
+                                                            maxSize: FxHomeHorizontalCellUX.siteFontSize)
+
+        descriptionLabel.textColor = viewModel.sponsor == nil ? .label : .secondaryLabel
```

Comment:
Good catch, it does. I was looking at the implementation and I needed two colors based on a story being sponsored or not, and the old implementation required me to store the story. 
But I took the opportunity to create the colors based on interface trait init. I'm curious what do you think about the solution. It should be pretty much the same thing without the notification setup.

Reply:
Oh I see! But without the notification setup we won't change the color on theme change? I'm saying this since `UIColor.theme.homePanel.recentlySavedBookmarkCellBackground` for the `contentView.backgroundColor` points to the dark theme and light theme. Easy way to test this is with the environmental overrides in Xcode

![Screen Shot 2022-05-26 at 10 58 29 AM](https://user-images.githubusercontent.com/11338480/170515231-6dc2cf3e-2a00-4fe0-a618-e1bb1f6e2202.png)

Reply:
Yes. Just checked and it adapts to the correct color.

https://user-images.githubusercontent.com/26011662/170693919-90654879-8f56-4398-91d7-3070054f4058.mov

Reply:
Awesome!